synapse/models/inet.py

```diff
@@ -35,6 +35,10 @@
 
 rfc6598 = ipaddress.IPv4Network('100.64.0.0/10')
 
+# defined from https://x.com/4A4133/status/1887269972545839559
+ja4_regex = r'^([tqd])(\d{2})([di])(\d{2})(\d{2})([a-z0-9]{2})_([0-9a-f]{12})_([0-9a-f]{12})$'
+ja4s_regex = r'^([tq])(\d{2})(\d{2})([a-z0-9]{2})_([0-9a-f]{4})_([0-9a-f]{12})$'
```

Comment:
Any concern about these being only lowercase?

Reply:
the types also specify lower=true on the strings. i believe all the values are lower case, but i'll check the corpus again.